### PR TITLE
MAINT: stats: fix _contains_nan on Pandas Series

### LINKED ISSUE #16211: BUG: Possible bug when using winsorize on pandas data instead of numpy data
### Describe your issue.

When `scipy.stats.mstats.winsorize` is used with a `nan_policy` of `omit` on a numpy array it behaves as expected. However, when used on Pandas data such as a series, all NaNs are converted to the maximum possible value. I believe this may be a bug, but could possibly be an enhancement if pandas data is never meant to be used. An example can be seen in the image below:

image

### Reproducing Code Example

```python
import numpy as np
import pandas as pd
from scipy.stats.mstats import winsorize

nans_10 = list(np.repeat(np.nan,10))
data = np.array(
	nans_10
	+ list(range(100))
	+ nans_10
)


print(f"Numpy array: {winsorize(data, (0.2, 0.2), nan_policy = 'omit')}")

print(f"Pandas series: {winsorize(pd.Series(data), (0.2, 0.2), nan_policy = 'omit')}")
```

### Error message

```shell
No error/warning message is given. The behavior is just unexpected. The pandas version used was 1.4.2.
```

### SciPy/NumPy/Python version information

1.8.0 1.21.6 sys.version_info(major=3, minor=10, micro=2, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
closes gh-16211

#### What does this implement/fix?
`_contains_nan`, the `stats` function that tries to beat `np.any(np.isnan(a))` at checking for `nan`s in an array, first tries `np.isnan(a.sum())`. If `a` is a Pandas Series, `sum` omits NaNs by default, so this doesn't work. This PR fixes the problem by skipping this strategy unless `a` is an `ndarray`.

#### Additional information
Ideas for tests and better ways of checking for NumPy array-ness are welcome. 
In the meantime, please test manually using the code in gh-16211. (CI doesn't have Pandas, to my knowledge, and I don't think it's important enough to write a class that behaves like them.)

### CODE REVIEW

**Review by tupui — APPROVED** [review]
LGTM, I don't think this needs an extra test (and we don't want pandas in the test suite). @tirthasheshpatel any comment before we merge?

In the end it will be hard to handle all cases (e.g. as seen in the other PR on NaNs). We should have some notes in the doc to incentivize users to deal with NaNs before feeding the array to SciPy.

**Comment by mdhaber** [comment]
Instead of this fix, @AtsushiSakai suggested ((https://github.com/AtsushiSakai/scipy/pull/2#discussion_r888866847)) that we could use the Python `sum` function instead of `np.sum` or the `.sum` method below, as this would not ignore NaNs in the calculation. It is a bit slower, but it would avoid needing this logic, and it would avoid creating a big array with `np.isnan` on Pandas objects. 

What do you all think of that? (I'm really not sure.)

**Comment by tupui** [comment]
I am not a NaN expert at all, so I am neutral here.

**Comment by mdhaber** [comment]
@tupui let's fix gh-16211 with this. @AtsushiSakai's suggestion might be better, but it needs a study of the cost/benefit, since it will slow the performance of `_contains_nan` for `ndarray`s somewhat, whereas this won't affect `ndarray`s at all. Let's merge this and if @AtsushiSakai is interested, I would be willing to review that would remove this patch and uses `sum` instead of `np.sum` below. It would help if the description of that PR were to include some performance information for different container types, dtypes, and sizes of input.

**Comment by tylerjereddy** [comment]
Small patch that cherry-picks gracefully and labeled as bug, so tentatively labeled for backport now.

**Pull request merged**
